test/unittesting/config_test.go

```diff
@@ -0,0 +1,1403 @@
+
+		if err != nil {
+			t.Error("could not find a supertokens instance")
+		}
+
+		if configValue.AppInfo.APIBasePath != nil {
+			assert.Equal(t, "/test", supertokensInstance.AppInfo.APIBasePath.GetAsStringDangerous())
+			assert.Equal(t, "/test1", supertokensInstance.AppInfo.WebsiteBasePath.GetAsStringDangerous())
+		} else {
+			assert.Equal(t, "/auth", supertokensInstance.AppInfo.APIBasePath.GetAsStringDangerous())
+			assert.Equal(t, "/auth", supertokensInstance.AppInfo.WebsiteBasePath.GetAsStringDangerous())
+		}
+		AfterEach()
+	}
+}
+
+func TestSuperTokensInitWithAbsenceOfCompulsoryInputInAppInfo(t *testing.T) {
+	configValues := []supertokens.TypeInput{
+		{
+			Supertokens: &supertokens.ConnectionInfo{
+				ConnectionURI: "http://localhost:8080",
+			},
+			AppInfo: supertokens.AppInfo{
+				AppName:       "SuperTokens",
+				WebsiteDomain: "supertokens.io",
+			},
+			RecipeList: []supertokens.Recipe{
+				session.Init(nil),
+			},
+		},
+		{
+			Supertokens: &supertokens.ConnectionInfo{
+				ConnectionURI: "http://localhost:8080",
+			},
+			AppInfo: supertokens.AppInfo{
+				APIDomain:     "api.supertokens.io",
+				WebsiteDomain: "supertokens.io",
+			},
+			RecipeList: []supertokens.Recipe{
+				session.Init(nil),
+			},
+		},
+		{
+			Supertokens: &supertokens.ConnectionInfo{
+				ConnectionURI: "http://localhost:8080",
+			},
+			AppInfo: supertokens.AppInfo{
+				AppName:   "SuperTokens",
+				APIDomain: "api.supertokens.io",
+			},
+			RecipeList: []supertokens.Recipe{
+				session.Init(nil),
+			},
+		},
+	}
+	for _, configValue := range configValues {
+		BeforeEach()
+		StartUpST("localhost", "8080")
+		err := supertokens.Init(configValue)
+		if err != nil {
+			errMessage := err.Error()
+			if configValue.AppInfo.AppName != "SuperTokens" {
+				assert.Equal(t, errMessage, "Please provide your appName inside the appInfo object when calling supertokens.init")
+			} else if configValue.AppInfo.APIDomain != "api.supertokens.io" {
+				assert.Equal(t, errMessage, "Please provide your apiDomain inside the appInfo object when calling supertokens.init")
+			} else if configValue.AppInfo.WebsiteDomain != "supertokens.io" {
+				assert.Equal(t, errMessage, "Please provide your websiteDomain inside the appInfo object when calling supertokens.init")
+			}
+		}
+		AfterEach()
+	}
+}
+
+func TestSuperTokensInitWith0RecipeModules(t *testing.T) {
+	configValue := supertokens.TypeInput{
+		Supertokens: &supertokens.ConnectionInfo{
+			ConnectionURI: "http://localhost:8080",
+		},
+		AppInfo: supertokens.AppInfo{
+			AppName:       "SuperTokens",
+			WebsiteDomain: "supertokens.io",
+			APIDomain:     "api.supertokens.io",
+		},
+		RecipeList: []supertokens.Recipe{},
+	}
+	BeforeEach()
+	StartUpST("localhost", "8080")
+	err := supertokens.Init(configValue)
+	if err != nil {
+		errorMessage := err.Error()
+		if errorMessage != "please provide at least one recipe to the supertokens.init function call" {
+			t.Errorf(err.Error())
```

Comment:
Somewhere you are using Errorf, and somewhere just Error. Is this intentional or a mistake?